src/extensions/replay/sessionrecording-utils.ts

```diff
@@ -165,27 +166,154 @@ export function truncateLargeConsoleLogs(_event: eventWithTime) {
-        const half = Math.floor(buffer.data.length / 2)
-        const firstHalf = buffer.data.slice(0, half)
-        const secondHalf = buffer.data.slice(half)
-        return [
-            splitBuffer({
-                size: estimateSize(firstHalf),
-                data: firstHalf,
-                sessionId: buffer.sessionId,
-                windowId: buffer.windowId,
-            }),
-            splitBuffer({
-                size: estimateSize(secondHalf),
-                data: secondHalf,
-                sessionId: buffer.sessionId,
-                windowId: buffer.windowId,
-            }),
-        ].flatMap((x) => x)
+        return sliceBuffer(buffer, sizeLimit)
+    } else if (buffer.size >= sizeLimit && buffer.data.length === 1) {
+        // we can maybe split up incremental snapshots, or directly edit data image urls here
+        const bufferedData = buffer.data[0]
+        if (
+            bufferedData.type === INCREMENTAL_SNAPSHOT_EVENT_TYPE &&
+            bufferedData.data.source === MUTATION_SOURCE_TYPE
+        ) {
+            // so at this point we know that the buffer is too large, and we have a single incremental snapshot
+            // it may be that a single item in the buffer is too large
+            // or that there are many small items of one or more types that end up being too large
+            // rrweb processes removes, then adds, then texts, the attributes,
+            // so we can split them in that order
+            const bufferedMutations = bufferedData.data as mutationData
+            const removes = sliceList(bufferedMutations.removes, sizeLimit)
+            const adds = sliceList(bufferedMutations.adds, sizeLimit)
+            const texts = sliceList(bufferedMutations.texts, sizeLimit)
+            const attributes = sliceList(bufferedMutations.attributes, sizeLimit)
+
+            // the incoming data has a single timestamp, so we need to adjust the timestamps of the split data,
+            // so we count how many children we have in total, and then we adjust the timestamp
+            // so that if there are 10 the first item is 9 milliseconds before the original timestamp
+            // and the final item has the original timestamp
+            const alteration =
+                countChildren(removes) + countChildren(adds) + countChildren(texts) + countChildren(attributes)
+            let timestampWiggleMarker = 1
+
+            return [
+                ...removes.map((remove) => ({
+                    size: estimateSize(remove),
+                    data: [
+                        {
+                            type: INCREMENTAL_SNAPSHOT_EVENT_TYPE,
+                            data: {
+                                source: MUTATION_SOURCE_TYPE,
+                                adds: [],
+                                texts: [],
+                                attributes: [],
+                                removes: remove,
```

Comment:
Tiny tiny nit but I feel like `removes` should be at the top given it happens first and the order is so important here. Wouldn't like any opportunity for someone to mix that up given we don't really have any guarantees over the ordering here